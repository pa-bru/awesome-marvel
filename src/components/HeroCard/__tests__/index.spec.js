import React from 'react'
import { shallow } from 'enzyme'
import { HeroCard } from '../'

// Given
const props = {
  urls: [
    {
      type: 'wiki',
      url: 'http://url.com',
    },
    {
      type: 'comiclink',
      url: 'http://url.com',
    },
  ],
  classes: {},
}

describe('HeroCard', () => {
  it('should render correctly', () => {
    // When
    const tree = shallow(<HeroCard {...props} />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render the extended version', () => {
    // When
    const tree = shallow(<HeroCard {...props} extended />)

    // Then
    expect(tree).toMatchSnapshot()
  })
})
