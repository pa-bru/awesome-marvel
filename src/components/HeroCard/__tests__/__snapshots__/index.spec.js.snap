// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard should render correctly 1`] = `
<WithStyles(Card)>
  <WithStyles(CardHeader)
    subheader="0 comics - 0 series - 0 events"
    title={
      <WithStyles(Typography)
        component="h2"
        variant="subheading"
      />
    }
  />
  <Link
    replace={false}
    to="/characters/undefined"
  >
    <WithStyles(CardMedia)
      image="."
    />
  </Link>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="p"
    >
      No description for this character. Sorry.
    </WithStyles(Typography)>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)
    disableActionSpacing={true}
  >
    <WithStyles(Button)
      color="primary"
      size="small"
    >
      <Link
        replace={false}
        to="/characters/undefined"
      >
        Detail
      </Link>
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      size="small"
    >
      <a
        href="http://url.com"
      >
        Wiki
      </a>
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      size="small"
    >
      <a
        href="http://url.com"
      >
        Comic Link
      </a>
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;

exports[`HeroCard should render the extended version 1`] = `
<WithStyles(Card)>
  <WithStyles(CardHeader)
    subheader="0 comics - 0 series - 0 events"
    title={
      <WithStyles(Typography)
        component="h1"
        variant="title"
      />
    }
  />
  <WithStyles(CardMedia)
    image="."
  />
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="p"
    >
      No description for this character. Sorry.
    </WithStyles(Typography)>
    <div>
      <WithStyles(Chip)
        label="0 comics"
      />
      <WithStyles(Chip)
        label="0 series"
      />
      <WithStyles(Chip)
        label="0 events"
      />
    </div>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)
    disableActionSpacing={true}
  >
    <WithStyles(Button)
      color="primary"
      size="small"
    >
      <a
        href="http://url.com"
      >
        Wiki
      </a>
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      size="small"
    >
      <a
        href="http://url.com"
      >
        Comic Link
      </a>
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
